src/FileSystemHandle.ts

```diff
@@ -0,0 +1,73 @@
+import { FileSystemHandleAdapter } from './interfaces'
+
+const kAdapter = Symbol('adapter')
+
+export class FileSystemHandle {
+  /** @internal */
+  [kAdapter]: FileSystemHandleAdapter
+
+  readonly kind: 'file' | 'directory'
+  readonly name: string
+
+  /** @deprecated */
+  get isFile(): any {
+    return this.kind === 'file'
+  }
+
+  /** @deprecated */
+  get isDirectory(): any {
+    return this.kind === 'directory'
+  }
+
+  constructor (adapter: FileSystemHandleAdapter) {
+    this.kind = adapter.kind
+    this.name = adapter.name
+    this[kAdapter] = adapter
+  }
+
+  async queryPermission (options: FileSystemHandlePermissionDescriptor = {}) {
+    const handle = this[kAdapter]
+    if (handle.queryPermission) {
+      return handle.queryPermission(options)
+    }
+
+    if (options.mode === 'read') {
+      return 'granted'
+    } else if (options.mode === 'readwrite') {
+      return handle.writable ? 'granted' : 'denied'
+    } else {
+      throw new TypeError(`Mode ${options.mode} must be 'read' or 'readwrite'`)
+    }
+  }
+
+  async requestPermission (options: FileSystemHandlePermissionDescriptor = {}) {
+    const handle = this[kAdapter]
+    if (handle.requestPermission) {
+      return handle.requestPermission(options)
+    }
+
+    if (options.mode === 'read') {
+      return 'granted'
+    } else if (options.mode === 'readwrite') {
+      return handle.writable ? 'granted' : 'denied'
+    } else {
+      throw new TypeError(`Mode ${options.mode} must be 'read' or 'readwrite'`)
```

Comment:
must have forgotten that mode can also be undefined, or rather have a default value

this works fine in chrome
```js
var e = await navigator.storage.getDirectory()
await e.requestPermission()
```

Reply:
Yeah, you are right. It wasn't clear from the API docs, but I see it  now in this [example](https://developer.mozilla.org/en-US/docs/Web/API/FileSystemHandle/queryPermission#examples). So ... `options = { mode: 'read' }` for both `queryPermission` and `requestPermission`?

Reply:
Fixed on my fork: https://github.com/use-strict/native-file-system-adapter/commit/817bdabe0521d7929854fb0b6b994829c816957d